.github/workflows/draw-zmk.yml

```
@@ -47,53 +52,78 @@ jobs:
+          fetch-depth: ${{ (inputs.amend_commit == true && 2) || 1 }}
+
+      - name: Install keymap-drawer (pypi)
+        if: inputs.install_branch == ''
+        run: python3 -m pip install keymap-drawer
+
+      - name: Install keymap-drawer (git)
+        if: inputs.install_branch != ''
+        run: python3 -m pip install "git+https://github.com/caksoylar/keymap-drawer.git@${{ inputs.install_branch }}"
+
+      - name: Draw keymaps
+        run: |
+          get_args() {
+              local keyboard=$2
+              eval set -- "$1"
+              for arg; do
+                  local key=${arg%%:*}
+                  local val=${arg#*:}
+                  if [ "$key" = "$keyboard" ]; then
+                      echo "$val"
+                      break
+                  fi
+              done
+          }
+
+          mkdir -p "${{ inputs.output_folder }}"
+
+          config_path="${{ inputs.config_path }}"
+          [ -e "$config_path" ] && config_arg=(-c "$config_path") || config_arg=()
+          for keymap_file in ${{ inputs.keymap_patterns }}; do
+              keyboard=$(basename -s .keymap "$keymap_file")
+              echo "INFO: drawing for $keyboard"
 
-        mkdir -p "${{ inputs.output_folder }}"
+              parse_args=$(get_args "${{ inputs.parse_args }}" "$keyboard")
+              echo "INFO:   got extra parse args: $parse_args"
+              draw_args=$(get_args "${{ inputs.draw_args }}" "$keyboard")
+              echo "INFO:   got extra draw args: $draw_args"
 
-        config_path="${{ inputs.config_path }}"
-        [ -e "$config_path" ] && config_arg=(-c "$config_path") || config_arg=()
-        for keymap_file in ${{ inputs.keymap_patterns }}; do
-            keyboard=$(basename -s .keymap "$keymap_file")
-            echo "INFO: drawing for $keyboard"
+              if [ -f "config/${keyboard}.json" ]; then
+                  echo "INFO:   found config/${keyboard}.json";
+                  draw_args+=" -j config/${keyboard}.json"
+              fi
 
-            parse_args=$(get_args "${{ inputs.parse_args }}" "$keyboard")
-            echo "INFO:   got extra parse args: $parse_args"
-            draw_args=$(get_args "${{ inputs.draw_args }}" "$keyboard")
-            echo "INFO:   got extra draw args: $draw_args"
+              keymap "${config_arg[@]}" parse -z "$keymap_file" $parse_args >"${{ inputs.output_folder }}/$keyboard.yaml" \
+              && keymap "${config_arg[@]}" draw "${{ inputs.output_folder }}/$keyboard.yaml" $draw_args >"${{ inputs.output_folder }}/$keyboard.svg" \
+              || echo "ERROR: parsing or drawing failed for $keyboard!"
+          done
 
-            if [ -f "config/${keyboard}.json" ]; then
-                echo "INFO:   found config/${keyboard}.json";
-                draw_args+=" -j config/${keyboard}.json"
-            fi
+      - name: Get last commit message
+        id: last_commit_message
+        if: ${{ inputs.amend_commit == true }}
```

```suggestion
        if: inputs.amend_commit == true
```
I don't think these are necessary here because `if` expects expressions already, like the install steps above.